---
title: "Genome Bioinformatics"
permalink: /drylab/
author_profile: false
sidebar:
  nav: sidebar-main
classes: wide
entries_layout: grid

feature_row:
  - image_path: /assets/images/datacleaning.jpg
    title: "Data Cleaning"
    url: "/data-cleaning/"
  - image_path: /assets/images/genomeassembly.png
    title: "Genome Assembly"
    url: "/assembly/"
  - image_path: /assets/images/geneannotation.jpg
    title: "Genome Annotation"
    url: "/annotation/"

header:
  overlay_image: /assets/images/home/cuke1.jpg

excerpt: "Genomes contain life's code. Here are my studies of a sea cucumber's code"
---

{% include feature_row %}

Here I have detailed the tools I've used and the process I went through to create the genome assembly, predict genes in it, and analyze the results. Additionally, I will try to give an explanation of how these tools function and why it's important to understand how they work. When reading these posts, keep in mind that it's a rare day when everything works the first time. More time is spent troubleshooting installation problems and learning how to use a tool then is spent actually using the tool. So if you find yourself struggling to get things working, rest assured, that's a battle we all fight everyday. 

Genomes do not spontaneously create themselves. At least in silico. Once raw sequence data is acquired, quality and content is checked. Then an assembly tool is selected and run. When assembled, the assembly quality is assessed. If it's "good enough" then repetitive element masking and gene prediction are performed. If everything checks out afterwards, the genome is ready for studying. Above are three pages containing content on the process for each of these steps. 




